pkg/backend_test.go

```diff
@@ -0,0 +1,115 @@
+package pkg
+
+import (
+	"strings"
+	"testing"
+)
+
+func checkError(got, expected error, t *testing.T) bool {
+	if (got != nil) != (expected != nil) {
+		t.Fatalf("expecting an error: '%+v', but got: '%+v'", expected, got)
+	}
+	if (got != nil) && (expected != nil) {
+		if got.Error() != expected.Error() {
+			t.Fatalf("\nexpected =\t'%+v'\ngot =\t'%+v'", expected, got)
+		} else {
+			return true
+		}
+	}
+	return false
+}
+func TestCompiler(gt *testing.T) {
+	swaggerContent := strings.ReplaceAll(`openapi: "3.0.0"
+components:
+	schemas:
+		products.inventory:
+			type: object
+			x-seal-actions:
+			- allow
+			- deny
+			x-seal-verbs:
+			- inspect
+			- use
+			- manage
+			x-seal-default-action: deny
+		dns.test:
```

Comment:
please remove domain specific things - do not use this type.  feel free to expand on products.inventory or company.personnel.  look at examples in docs/source/grammar/grammar.md or docs/source/examples/simple/products.inventory.swagger